plugins/org.yakindu.sct.doc.user/src/user-guide/sctunit.textile

```diff
@@ -5,370 +5,941 @@ p.
+
+The assertion fails if the state machine is not active (see above) or it has at least one active state that is not a final state or composite state.
+
+==</div>==
+
+==<!-- End sctunit_keyword_is_final -->==
+==<!-- End sctunit_keyword_is_active -->==
+
+
+
+==<!-- Start sctunit_keyword_mock -->==
+
+h4(#sctunit_mocking_an_operation_call). Mocking an operation call
+
+The _mock_ statement acts as a placeholder for an operation call and its return value.
+
+==<div class="example">==
+
+Consider a complex operation _getNeutronFlux_, which takes a real value as an argument and returns a real value as a result. During _semantic unit testing_ of your statechart – as opposed to _integration testing_ –, you won't want to integrate the operation into your testing environment. Aside from that, SCTUnit tests are executed in YAKINDU Statechart Tools' simulation mode, where you cannot call operations anyway.
+
+Your statechart, however, depends on _getNeutronFlux_ returning actual results, as can be seen in the simple model shown in figure "&quot;Neutron flux statechart&quot;":#sctunit_fig_neutron_flux_statechart:
+
+p(#sctunit_fig_neutron_flux_statechart). 
+
+!images/sctunit_neutron_flux_010_statechart.png(Neutron flux statechart)!
+
+p=. Neutron flux statechart
+
+While *State_A* is active, the state machine will call _getNeutronFlux(p)_ during each run-to-completion step and, depending on the results, will take the transition to *State_B* (or not).
+
+==</div>==
+
+However, how could you test the behaviour of your statechart if you cannot call an actual operation and retrieve its results?
+
+That's what the _mock_ statement is for. It is a makeshift that mimics an actual operation call by two mechanisms:
+# It creates a static mapping from a specific operation call with a specific list of parameter values to a specific return value.
+# If that operation is called with exactly that specific list of parameter values the caller receives the mapped return value as a result.
+
+A test can create an arbitrary number of such mappings, i.e. it can use the _mock_ statement multiple times for multiple operations, or for multiple parameter list of the same operation.
+
+==<div class="example">==
+
+Consider the following test:
+
+bc. 
+    @Test
+    operation testNeutronFlux() {
+        mock getNeutronFlux(12.0) returns (1000000000.0)
+        mock getNeutronFlux(18.0) returns (4.3)
+        enter
+        p = 18.0
+        proceed 1 cycle
+        assert active(neutronFlux.main_region.State_A)
+        p = 12.0
+        proceed 1 cycle
+        assert active(neutronFlux.main_region.State_B)
+    }
+
+The effect of the two _mock_ statements is as follows:
+* If _getNeutronFlux(12.0)_ is called, the return value is 1000000000.0.
+* If _getNeutronFlux(18.0)_ is called, the return value is 4.3.
+* If _getNeutronFlux_ is called with any other parameter value, the return value is undefined.
```

Comment:
And the test will fail and throw an exception cause the return value and therefore the behaviour of the operation is undefined.